content/en/blog/_posts/2021-09-07-data-duplication-in-data-heavy-k8s-env.md

```diff
@@ -0,0 +1,221 @@
+find volume Id (vol-00c7ec…) by the CSIVolumeName tag first.
+
+<img src="/images/blog/2021-09-07-data-duplication-in-data-heavy-k8s-env/identify-volume-aws.png" width="100%" alt="Identify volume AWS" />
+
+And now it’s time to click Actions -> Create Snapshot, or use ```aws cli```.
+
+```terminal
+aws ec2 create-snapshot --volume-id vol-00c7ecd873c6fb3ec --description 'postgresql-orders-db-gold' --tag-specifications 'ResourceType=snapshot,Tags=[{Key=purpose,Value=prod},{Key=costcenter,Value=123}]'
+```
+
+## Step 3: Get your Disk Snapshot ID
+
+In AWS, the aws cli command will output:  ```"SnapshotId": "snap-09ed24a70bc19bbe4"```
+
+In the console it will be displayed like this:
+
+<img src="/images/blog/2021-09-07-data-duplication-in-data-heavy-k8s-env/get-snapshot-id-aws.png" width="100%" alt="find snapshot id AWS" />
+
+If you’re using the GCP Console, you can click on the REST response and copy the ID “6645363163809389170.”
+
+<img src="/images/blog/2021-09-07-data-duplication-in-data-heavy-k8s-env/get-snapshot-id-gcp.png" width="100%" alt="find snapshot id GCP" />
+
+You can also get it from the gcloud command by the Snapshot’s given name:
+
+```terminal
+gcloud compute snapshots --project=staging-eu-castai-vt5hy2 describe postgresql-orders-db-gold | grep id:
+```
+
+## Step 4: Create a development environment for each team
+
+Now I have my Golden Snapshot, which is immutable data. Each team will get a copy 
+of this data, and team members can modify it as they see fit, given that a new EBS/persistent 
+disk will be created for each team.
+
+Below I will add K8s YAML for each namespace, replacing the namespace “sandbox01 -> sandbox42”
+names with yq, sed, or templating tools like Kustomize or using variables in the CI/CD pipeline:
+
+```yaml
+apiVersion: snapshot.storage.k8s.io/v1
+kind: VolumeSnapshotContent
+metadata:
+ name: postgresql-orders-db-sandbox01
+ namespace: sandbox01
+spec:
+ deletionPolicy: Retain
+ driver: storage.csi.cast.ai
+ source:
+   snapshotHandle: 'gcp/projects/staging-eu-castai-vt5hy2/global/snapshots/6645363163809389170'
+   #snapshotHandle: 'snap-07ff83d328c981c98' #for AWS
+ volumeSnapshotRef:
+   kind: VolumeSnapshot
+   name: postgresql-orders-db-snap
+   namespace: sandbox01
+---
+apiVersion: snapshot.storage.k8s.io/v1
+kind: VolumeSnapshot
+metadata:
+ name: postgresql-orders-db-snap
+ namespace: sandbox01
+spec:
+ source:
+   volumeSnapshotContentName: postgresql-orders-db-sandbox01
+   ```
+
+VolumeSnapshotContent (VSC) is not a namespaced K8s object. But I will create an 
+instance of this object for every namespace, so the name needs to be unique (hence the namespace in VSC name).
+
+Now it’s time to replace the driver field with the CSI driver installed in the K8s cluster.
+The CAST AI CSI driver supports all of the major cloud services so this YAML will work on AWS,
+GCP, and probably Azure (though Azure has been lagging on their cloud storage drivers for K8s)
+on the CAST AI K8s cluster.
```

Comment:
We avoid introducing new commercial dependencies without a very compelling reason. This would require review from someone more senior in sig docs. 

I do not think this use case is compelling. If the snapshot is on AWS, I'm going to use the AWS driver.

Reply:
Fair point